src/ParamHandler.ts

```diff
@@ -104,7 +104,7 @@ export class ParamHandler {
     private parseValue(value: any, paramMetadata: ParamMetadata) {
         try {
             const parseValue = typeof value === "string" ? JSON.parse(value) : value;
-            if (paramMetadata.format !== Object && paramMetadata.format && this.driver.useClassTransformer) {
+            if (!(value instanceof paramMetadata.format) && paramMetadata.format !== Object && paramMetadata.format && this.driver.useClassTransformer) {
```

Comment:
ohhh this code gets messier... can you add comment why you added this line, also make sure to include test that coverts this scenario